From 27e2e67ead167c8c4b3ddc81c97ab0d0e03abc0d Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Tue, 27 Apr 2021 14:42:36 +0700
Subject: [PATCH 364/364] arm64: dts: renesas: r8a774e1: Un-tie IPMMU with
 devices

This commit un-tie devices with IPMMU, also disable IPMMUs
to support similar to r8a7795 in v4.14.75-ltsi/rcar-3.9.6.
Also it will be consistent with all RZ/G2 SoCs.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774e1.dtsi | 59 +++++++------------------------
 1 file changed, 13 insertions(+), 46 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a774e1.dtsi b/arch/arm64/boot/dts/renesas/r8a774e1.dtsi
index 4bb140b..f633d84 100644
--- a/arch/arm64/boot/dts/renesas/r8a774e1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774e1.dtsi
@@ -1033,14 +1033,6 @@
 			resets = <&cpg 219>;
 			#dma-cells = <1>;
 			dma-channels = <16>;
-			iommus = <&ipmmu_ds0 0>, <&ipmmu_ds0 1>,
-				 <&ipmmu_ds0 2>, <&ipmmu_ds0 3>,
-				 <&ipmmu_ds0 4>, <&ipmmu_ds0 5>,
-				 <&ipmmu_ds0 6>, <&ipmmu_ds0 7>,
-				 <&ipmmu_ds0 8>, <&ipmmu_ds0 9>,
-				 <&ipmmu_ds0 10>, <&ipmmu_ds0 11>,
-				 <&ipmmu_ds0 12>, <&ipmmu_ds0 13>,
-				 <&ipmmu_ds0 14>, <&ipmmu_ds0 15>;
 		};
 
 		dmac1: dma-controller@e7300000 {
@@ -1075,14 +1067,6 @@
 			resets = <&cpg 218>;
 			#dma-cells = <1>;
 			dma-channels = <16>;
-			iommus = <&ipmmu_ds1 0>, <&ipmmu_ds1 1>,
-				 <&ipmmu_ds1 2>, <&ipmmu_ds1 3>,
-				 <&ipmmu_ds1 4>, <&ipmmu_ds1 5>,
-				 <&ipmmu_ds1 6>, <&ipmmu_ds1 7>,
-				 <&ipmmu_ds1 8>, <&ipmmu_ds1 9>,
-				 <&ipmmu_ds1 10>, <&ipmmu_ds1 11>,
-				 <&ipmmu_ds1 12>, <&ipmmu_ds1 13>,
-				 <&ipmmu_ds1 14>, <&ipmmu_ds1 15>;
 		};
 
 		dmac2: dma-controller@e7310000 {
@@ -1117,14 +1101,6 @@
 			resets = <&cpg 217>;
 			#dma-cells = <1>;
 			dma-channels = <16>;
-			iommus = <&ipmmu_ds1 16>, <&ipmmu_ds1 17>,
-				 <&ipmmu_ds1 18>, <&ipmmu_ds1 19>,
-				 <&ipmmu_ds1 20>, <&ipmmu_ds1 21>,
-				 <&ipmmu_ds1 22>, <&ipmmu_ds1 23>,
-				 <&ipmmu_ds1 24>, <&ipmmu_ds1 25>,
-				 <&ipmmu_ds1 26>, <&ipmmu_ds1 27>,
-				 <&ipmmu_ds1 28>, <&ipmmu_ds1 29>,
-				 <&ipmmu_ds1 30>, <&ipmmu_ds1 31>;
 		};
 
 		ipmmu_ds0: iommu@e6740000 {
@@ -1133,6 +1109,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 0>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_ds1: iommu@e7740000 {
@@ -1141,6 +1118,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 1>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_hc: iommu@e6570000 {
@@ -1149,6 +1127,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 2>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_mm: iommu@e67b0000 {
@@ -1158,6 +1137,7 @@
 				     <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_mp0: iommu@ec670000 {
@@ -1166,6 +1146,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 4>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_pv0: iommu@fd800000 {
@@ -1174,6 +1155,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 6>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_pv1: iommu@fd950000 {
@@ -1182,6 +1164,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 7>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_pv2: iommu@fd960000 {
@@ -1190,6 +1173,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 8>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_pv3: iommu@fd970000 {
@@ -1198,6 +1182,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 9>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_vc0: iommu@fe6b0000 {
@@ -1206,6 +1191,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 12>;
 			power-domains = <&sysc R8A774E1_PD_A3VC>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_vc1: iommu@fe6f0000 {
@@ -1214,6 +1200,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 13>;
 			power-domains = <&sysc R8A774E1_PD_A3VC>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_vi0: iommu@febd0000 {
@@ -1222,6 +1209,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 14>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_vi1: iommu@febe0000 {
@@ -1230,6 +1218,7 @@
 			renesas,ipmmu-main = <&ipmmu_mm 15>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
+			status = "disabled";
 		};
 
 		ipmmu_vp0: iommu@fe990000 {
@@ -1288,7 +1277,6 @@
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			resets = <&cpg 812>;
 			phy-mode = "rgmii";
-			iommus = <&ipmmu_ds0 16>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 			status = "disabled";
@@ -2022,14 +2010,6 @@
 			resets = <&cpg 502>;
 			#dma-cells = <1>;
 			dma-channels = <16>;
-			iommus = <&ipmmu_mp0 0>, <&ipmmu_mp0 1>,
-				 <&ipmmu_mp0 2>, <&ipmmu_mp0 3>,
-				 <&ipmmu_mp0 4>, <&ipmmu_mp0 5>,
-				 <&ipmmu_mp0 6>, <&ipmmu_mp0 7>,
-				 <&ipmmu_mp0 8>, <&ipmmu_mp0 9>,
-				 <&ipmmu_mp0 10>, <&ipmmu_mp0 11>,
-				 <&ipmmu_mp0 12>, <&ipmmu_mp0 13>,
-				 <&ipmmu_mp0 14>, <&ipmmu_mp0 15>;
 		};
 
 		audma1: dma-controller@ec720000 {
@@ -2064,14 +2044,6 @@
 			resets = <&cpg 501>;
 			#dma-cells = <1>;
 			dma-channels = <16>;
-			iommus = <&ipmmu_mp0 16>, <&ipmmu_mp0 17>,
-				 <&ipmmu_mp0 18>, <&ipmmu_mp0 19>,
-				 <&ipmmu_mp0 20>, <&ipmmu_mp0 21>,
-				 <&ipmmu_mp0 22>, <&ipmmu_mp0 23>,
-				 <&ipmmu_mp0 24>, <&ipmmu_mp0 25>,
-				 <&ipmmu_mp0 26>, <&ipmmu_mp0 27>,
-				 <&ipmmu_mp0 28>, <&ipmmu_mp0 29>,
-				 <&ipmmu_mp0 30>, <&ipmmu_mp0 31>;
 		};
 
 		xhci0: usb@ee000000 {
@@ -2178,7 +2150,6 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			resets = <&cpg 314>;
-			iommus = <&ipmmu_ds1 32>;
 			status = "disabled";
 		};
 
@@ -2191,7 +2162,6 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			resets = <&cpg 313>;
-			iommus = <&ipmmu_ds1 33>;
 			status = "disabled";
 		};
 
@@ -2204,7 +2174,6 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			resets = <&cpg 312>;
-			iommus = <&ipmmu_ds1 34>;
 			status = "disabled";
 		};
 
@@ -2217,7 +2186,6 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			resets = <&cpg 311>;
-			iommus = <&ipmmu_ds1 35>;
 			status = "disabled";
 		};
 
@@ -2229,7 +2197,6 @@
 			clocks = <&cpg CPG_MOD 815>;
 			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
 			resets = <&cpg 815>;
-			iommus = <&ipmmu_hc 2>;
 			status = "disabled";
 		};
 
-- 
2.7.4

